src/app/util/ContentApp.cpp

```diff
@@ -0,0 +1,164 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file Contains shell commands for a commissionee (eg. end device) related to commissioning.
+ */
+
+#include <app-common/zap-generated/attribute-id.h>
+#include <app-common/zap-generated/cluster-id.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/util/ContentAppPlatform.h>
+#include <cstdio>
+#include <inttypes.h>
+#include <lib/core/CHIPCore.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/CHIPArgParser.hpp>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/ZclString.h>
+#include <platform/CHIPDeviceLayer.h>
+
+using namespace chip;
+using namespace chip::AppPlatform;
+
+namespace chip {
+namespace AppPlatform {
+
+#define ZCL_DESCRIPTOR_CLUSTER_REVISION (1u)
+#define ZCL_APPLICATION_BASIC_CLUSTER_REVISION (1u)
+
+EmberAfStatus ContentApp::HandleReadAttribute(ClusterId clusterId, chip::AttributeId attributeId, uint8_t * buffer,
+                                              uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "Read Attribute for device %s cluster %d attribute=%d)",
+                    GetApplicationBasic()->GetApplicationName(), static_cast<uint16_t>(clusterId),
+                    static_cast<uint16_t>(attributeId));
+
+    EmberAfStatus ret = EMBER_ZCL_STATUS_FAILURE;
+    if (clusterId == ZCL_APPLICATION_BASIC_CLUSTER_ID)
+    {
+        ret = GetApplicationBasic()->HandleReadAttribute(attributeId, buffer, maxReadLength);
+    }
+    if (clusterId == ZCL_ACCOUNT_LOGIN_CLUSTER_ID)
+    {
+        ret = GetAccountLogin()->HandleReadAttribute(attributeId, buffer, maxReadLength);
+    }
+    return ret;
+}
+
+EmberAfStatus ContentApp::HandleWriteAttribute(ClusterId clusterId, chip::AttributeId attributeId, uint8_t * buffer)
+{
+    ChipLogProgress(DeviceLayer, "Read Attribute for device %s cluster %d attribute=%d)",
+                    GetApplicationBasic()->GetApplicationName(), static_cast<uint16_t>(clusterId),
+                    static_cast<uint16_t>(attributeId));
+
+    EmberAfStatus ret = EMBER_ZCL_STATUS_FAILURE;
+
+    if (clusterId == ZCL_APPLICATION_BASIC_CLUSTER_ID)
+    {
+        ret = GetApplicationBasic()->HandleWriteAttribute(attributeId, buffer);
+    }
+    if (clusterId == ZCL_ACCOUNT_LOGIN_CLUSTER_ID)
+    {
+        ret = GetAccountLogin()->HandleWriteAttribute(attributeId, buffer);
+    }
+    return ret;
+}
+
+EmberAfStatus ApplicationBasic::HandleReadAttribute(chip::AttributeId attributeId, uint8_t * buffer, uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "ApplicationBasic::HandleReadAttribute: attrId=%d, maxReadLength=%d",
+                    static_cast<uint16_t>(attributeId), maxReadLength);
+
+    if ((attributeId == ZCL_APPLICATION_VENDOR_NAME_ATTRIBUTE_ID) && (maxReadLength == 32))
```

Comment:
Please use the namespaced attribute ids.

Reply:
Agreed, now that we have it...

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/12969